sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/TaskUtils.java

```diff
@@ -531,7 +531,9 @@ public static boolean needsRecovery(TaskSpec taskSpec, TaskStatus taskStatus) {
         // Tasks with a goal state of finished should never leave the purview of their original
```

Comment:
Nit, but I find this comment pretty obtuse. Even knowing how things work, it took me a second to realize that finished tasks "should never leave the purview of their original plan" just means that the plan responsible for finished tasks is going to keep restarting the task until it reaches its goal state, and then since it's done it should never need recovery.